guide/configuration.md

```diff
@@ -295,9 +293,86 @@ We can also configure some additional compilation and linking options through `-
 e.g:
 
 ```bash
-$ xmake f -p cross --sdk=/usr/toolsdk --cflags="-DTEST -I/xxx/xxx" --ldflags="-lpthread"
+$ xmake f -p cross --sdk=/usr/toolsdk --cflags="-DTEST -I/xxx/xxx" --ldflags="-lpthread" 
 ```
 
+### Step by Step Cross Compilation Tutorial
+
+#### Introduction
+
+Cross-compilation is the process of compiling code for a target architecture that differs from the one on your local machine. To do this, you must compile your source files for the desired target architecture and link them against the appropriate system libraries, typically provided through an SDK or sysroot.
+
+#### Why Cross Compile?
+
+* Native compilation isn’t always feasible, especially in fields like embedded systems development
+
+* Cross-compiling on a powerful development machine is often much faster than compiling directly on the target device
+
+* Relying solely on native compilation is impractical—there are many operating systems and architectures, and you can’t realistically own and maintain 20 different machines just to support them all
+
+#### How do you cross compile on xmake?
+
+1. Use a toolchain that supports cross-compilation.
+
+2. Set the platform to `cross` and specify the target triplet using the `--cross` parameter
+
+3. Provide the appropriate SDK or sysroot using the `--sdk` parameter
+
+#### Example
+
+In this example, you'll use cross compilation to run a hello world on a debian mint virtual machine. I'm assuming your host compiler is on windows.
+
+##### Getting the sysroot
+
+1. Install wsl, docker, and rancher desktop
+
+2. Pull the `debian:bookworm` image, update the package list, and install `g++`, which includes the C++ standard library we'll be linking against
+   
```

Comment:
I don't think the configuration details of these environments should be documented, which will only mislead users to install wsl, debian and docker.

But many times the user's environment does not require these, just download the corresponding cross-compilation tool chain.